pages/api/network/index.ts

```diff
@@ -195,19 +195,45 @@ async function post(req: NextApiRequest, res: NextApiResponse) {
     //TODO: move tokens logic to new endpoint   
     if(allowedTokens?.allowedTransactions?.length > 0){
       for (const token of allowedTokens.allowedTransactions) {
-        await Database.networkTokens.create({
-          networkId: network.id,
-          tokenId: token.id
+        const [networkToken, created]= await Database.networkTokens.findOrCreate({
```

Comment:
This file has enough repeated code that it becomes a pain whenever we need to change something (as seen on this PR).

_I would_ move the "loop, find and create" logic to a closure (or helper) function and then call it from here with the needed parameters. Effectively reducing the complexity and work we need to do when this happens;

Reply:
initially that's what I did but I had some problems I'll try again and see if the result holds